Allow reading of version information from env-vars

MKodde opened the pull request:
At this point,  you can set three env-vars that are evaluated to display on the /info endpoint.

If these variables are not present, the default behavior triggers (determining the version info based on the path information.

1. By default this is docker friendly (as our OpenContext images ship with these env-vars)
2. The fallback is usefull for when you install the application on bare metal, and unzip our openconext tarbals.

MKodde committed:
Introduce envvar support for info

MKodde commented on src/DependencyInjection/Compiler/HealthCheckPass.php:
```diff
@@ -38,7 +38,7 @@ public function process(ContainerBuilder $container)
         $definition = $container->findDefinition('openconext.monitor.health_check_chain');
 
-        // find all service IDs with the app.mail_transport tag
+        // find all service IDs with the openconext.monitor.health_check tag
```
Boyscout rule :evergreen_tree:

MKodde commented on phpmd.xml:
```diff
@@ -30,6 +30,8 @@
     <rule ref="rulesets/naming.xml">
         <exclude name="ShortVariable" />
         <exclude name="LongVariable" />
+        <exclude name="LongClassName" />
+        <exclude name="ShortClassName" />
```
Conform to the latest QA tool PHPMD ruleset we use in other OpenConext apps

MKodde commented on composer.json:
```diff
@@ -37,11 +37,11 @@
-      "phpunit": "vendor/bin/phpunit --coverage-text"
-      },
+      "phpunit": "vendor/bin/phpunit --coverage-text",
+      "phpmd": "vendor/bin/phpmd src text phpmd.xml --exclude 'src/Tests/*'"
```
Exclude the Unit tests from php md sniffing.

MKodde committed:
Raise Doctrine ORM support

To now also include support for doctrine 3

MKodde committed:
Add return types preventitavely

parijke commented on src/Value/BuildInformation.php:
```diff
@@ -0,0 +1,43 @@
+
+use JsonSerializable;
+
+interface BuildInformation extends JsonSerializable
```
suffix Interface or not?

MKodde replied:
I'm not going to add the suffix here.

parijke commented on src/Value/BuildPath.php:
```diff
@@ -26,45 +26,27 @@
-    public function __construct($path, $version = '', $revision = '')
+    public function __construct(
+        private readonly string $path,
+        private readonly string $version = '',
```
Why not initialise all vars?

MKodde replied:
The path is the one param that MUST be passed to the constructor. That's why I'm not alllowing a default value.

parijke commented on src/Value/Information.php:
```diff
@@ -26,48 +26,31 @@
-    public static function buildFrom(BuildPath $buildPath, string $environment, bool $debuggerEnabled, array $systemInfo): Information
-    {
+    public static function buildFrom(
+        BuildInformation $buildPath,
```
Beware of not initialising vars (see my article as well)

MKodde replied:
I only want to set default values when they are usefull. In this case, all vars are initialized during construction time. None of the params are optional. So this should never lead to problems right?

MKodde committed:
Tweak BuildPath and Information value objects

- Removed a redundant assertion from Information
- Made the BuildPath class readonly
